### DISCUSSION TITLE
react-form-form@6.9.6 depends on react@"^16.8.0". When will there be a version for react@17?

### DISCUSSION THREAD

**Discussion by ar4hc** [react-hook-form/react-hook-form#3266] [General] [answered]
This requirement seems to be new.
What broke react-form-form in react@17?
How long until there will be a version compatible with with react@17?

Not that i worry to much, but would be nice to know...

Thanks,

**Comment by bluebill1049** [comment]
it's compatible with react 17. react 16.8+
it already passes the integration test: https://github.com/react-hook-form/react-hook-form/blob/master/app/package.json

**Reply by ar4hc** [reply]
ok, then i got a strange error:
```
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.8.0" from react-hook-form@6.9.6
npm ERR! node_modules/react-hook-form
npm ERR!   react-hook-form@"^6.9.6" from the root project
npm ERR!
```
i read that as forms doesn't want to play with react 17... Am i wrong?

remove package-locks and node-modules dir, same error...

???

PS: Thanks for the quick answer.

**Comment by bluebill1049** [comment]
not sure about your setup.

- clone this and run it: https://github.com/react-hook-form/react-hook-form/tree/master/app
- running 17 at CSB: https://codesandbox.io/s/shy-thunder-63126?file=/src/App.tsx

**Reply by ar4hc** [reply]
first  gave the same error:
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: react-hook-form-automation-server@0.1.0
npm ERR! Found: react@17.0.1
npm ERR! node_modules/react
npm ERR!   react@"^17.0.1" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.8.0" from react-hook-form@6.9.6
npm ERR! node_modules/react-hook-form
npm ERR!   react-hook-form@"^6.9.6" from the root project
npm ERR!
...
```
local on macOS 10.15.7 (19H2) with npm 7.0.3... 
:(

**Comment by ar4hc** [comment]
i used `npm i --legacy-peer-deps`and it seems to work.

my projects package.json looks like this:
```
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "react-hook-form": "^6.9.6",
    "react-number-format": "^4.4.1",
    "react-router": "^5.2.0",
    "react-router-dom": "^5.2.0",
    "react-scripts": "^4.0.0",
    "typescript": "^4.0.3"
```
and dev like:
```
  "devDependencies": {
    "@hookform/devtools": "^2.2.1",
    "canvas": "^2.6.1",
    "eslint": "^7.12.0",
    "eslint-config-prettier": "^6.14.0",
    "eslint-plugin-prettier": "^3.1.4",
    "eslint-plugin-react": "^7.21.5",
    "eslint-plugin-react-hooks": "^4.2.0",
    "notistack": "^1.0.1",
    "prettier": "^2.1.2",
    "prettier-eslint": "^11.0.0",
    "prop-types": "^15.7.2"
  }
```
i don't see any significant difference here...

rm package-lock and node-modules, fought a bit with a more picky eslint, but the app itself seems to work fine.

any ideas...?

**Comment by ar4hc** [comment, marked as answer]
looks like its related to this [npm bug](https://github.com/npm/cli/issues/1989) , i'll try yarn...

**Reply by bluebill1049** [reply]
I am adding this anyhow: https://github.com/react-hook-form/react-hook-form/pull/3273 :)
